cosmotech/coal/cli/commands/tdl_load_files.py

```diff
@@ -47,28 +56,45 @@
               envvar="CSM_DATASET_ABSOLUTE_PATH",
               show_envvar=True,
               required=True)
-def tdl_load_file(
+def tdl_load_files(
     organization_id,
     workspace_id,
+    scenario_id,
     runner_id,
     directory_path
 ):
     """Query a twingraph and loads all the data from it
 
 Will create 1 csv file per node type / relationship type
 
-The twingraph must have been populated using the "tdl-send-file" command for this to work correctly
+The twingraph must have been populated using the "tdl-send-files" command for this to work correctly
 
 Requires a valid connection to the API to send the data
     """
 
     api_client, connection_type = get_api_client()
     api_ds = DatasetApi(api_client)
-    api_runner = RunnerApi(api_client)
 
-    runner_info = api_runner.get_runner(organization_id,
-                                        workspace_id,
-                                        runner_id)
+    runner_info = None
+    if scenario_id:
+        if runner_id:
+            LOGGER.info(f"Both scenario ({scenario_id}) and runner ({runner_id}) id are defined; defaulting to scenario.")
+        scenario_api = ScenarioApi(api_client)
+        runner_info = scenario_api.find_scenario_by_id(
+            organization_id,
+            workspace_id,
+            scenario_id,
+        )
+    elif runner_id:
+        api_runner = RunnerApi(api_client)
+        runner_info = api_runner.get_runner(
+            organization_id,
+            workspace_id,
+            runner_id,
+        )
+    else:
+        LOGGER.error(f"Neither scenario nor runner id is defined.")
+        raise click.Abort()
 
     if len(runner_info.dataset_list) != 1:
         LOGGER.error(f"Runner {runner_id} is not tied to a single dataset")
```

Comment:
If using scenarios, and if the provided scenario does not have any datasets, this error message will show `None` instead of the scenario id (or  complain that `runner_id` is not defined) :slightly_smiling_face: